crates/core/src/client_events.rs

```diff
@@ -316,6 +320,113 @@ pub(crate) mod test {
         }
     }
 
+    pub struct NetworkEventGenerator<R = rand::rngs::SmallRng> {
+        id: PeerId,
+        memory_event_generator: MemoryEventsGen<R>,
+        ws_client: Arc<Mutex<WebSocketStream<MaybeTlsStream<TcpStream>>>>,
+        rng: Option<R>,
```

Comment:
does this need to be an option if is always set?

Reply:
fixed